scipy/optimize/optimize.py

```diff
@@ -48,7 +48,9 @@
                    'maxiter': 'Maximum number of iterations has been '
                               'exceeded.',
                    'pr_loss': 'Desired error not necessarily achieved due '
-                              'to precision loss.'}
+                              'to precision loss.',
+                   'nonfinite': 'Optimization encountered non-finite numerical'
```

Comment:
For purposes of clarity is it worth having a message for NaN and a message for Inf? Some optimizers can deal with Inf, but can't deal with NaN. The changes here are mainly for NaN.

If the message is intended for NaN, then perhaps the message should be `Optimization encountered NaN numerical values`, and the key changed to `'nan'`.

Reply:
Changed the wording to say NaN, since that's what's tested.

Solvers will probably tolerate inf better, since it behaves similarly to other floats in comparisons.